shared-libs/rules-engine/src/rules-state-store.js

```diff
@@ -108,6 +110,7 @@ const self = {
         rulesConfigHash,
         contactState: {},
         targetState: targetState.createEmptyState(settings.targets),
+        monthStartDate: settings.monthStartDate,
```

Comment:
You may be able to avoid coupling these libraries by passing a callback like `getCurrentReportingInterval` into rules-engine

Reply:
That doesn't really change the coupling - all three libraries (cht-core, the rules-engine, and the calendar-interval) will still be coupled. You're just moving a coupling from the rules engine to the cht core, and making the API more complicated. I prefer it as it is.